atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -209,38 +219,24 @@ private static boolean keyspaceAlreadyExists(InetSocketAddress host, CassandraKe
         }
     }
 
-    private static void attemptToCreateKeyspaceOnHost(InetSocketAddress host, CassandraKeyValueServiceConfig config)
+    private static boolean attemptToCreateKeyspaceOnHost(InetSocketAddress host, CassandraKeyValueServiceConfig config)
             throws TException {
-        CassandraClient client = CassandraClientFactory.getClientInternal(host, config);
-        KsDef ks = new KsDef(config.getKeyspaceOrThrow(), CassandraConstants.NETWORK_STRATEGY,
-                ImmutableList.of());
-        checkAndSetReplicationFactor(
-                client,
-                ks,
-                true,
-                config);
-        ks.setDurable_writes(true);
-        client.system_add_keyspace(ks);
-        log.info("Created keyspace: {}", UnsafeArg.of("keyspace", config.getKeyspaceOrThrow()));
-        CassandraKeyValueServices.waitForSchemaVersions(
-                config,
-                client,
-                "(adding the initial empty keyspace)",
-                true);
-    }
-
-    private static boolean attemptedToCreateKeyspaceTwice(InvalidRequestException ex) {
-        String exceptionString = ex.toString();
-        if (exceptionString.contains("case-insensitively unique")) {
-            String[] keyspaceNames = StringUtils.substringsBetween(exceptionString, "\"", "\"");
-            if (keyspaceNames.length == 2 && keyspaceNames[0].equals(keyspaceNames[1])) {
-                return true;
-            }
```

Comment:
This case is replaced by the call to `keyspaceAlreadyExists()` too - it incurs an extra network call, but this case is uncommon.